mindmeld/text_preparation/normalizers.py

```diff
@@ -195,28 +212,150 @@ def normalize(self, text):
         return unicodedata.normalize(self.normalization_type, text)
 
 
+class Lowercase(Normalizer):
+    """Lowercase Normalizer Class."""
+
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        return text.lower()
+
+
+class RegexNormalizerRule(Normalizer):
+    def __init__(self, pattern: str, replacement: str):
+        """Creates a RegexNormalizerRule instance."""
+        self.pattern = pattern
+        self.replacement = replacement
+        self._expr = re.compile(self.pattern)
+
+    def normalize(self, s):
+        return self._expr.sub(self.replacement, s)
+
+
+class RegexNormalizerRuleFactory:
+
+    exception_chars = r"\@\[\]'"
```

Comment:
Could you clarify why we need this as a class var? Why not just do `def __init__` and place `exception_chars ` as an instance var?

Reply:
I went with a class variable since the Factory only has static methods. Moreover, by updating the class variable the change persists throughout the whole system. However, since Normalizers are only made once anyways, I can switch this to instantiate the Factory instead.

Reply:
Added a comment here